include/triton/Tools/DiagEmitter.hpp

```diff
@@ -0,0 +1,51 @@
+#ifndef TRITON_TOOLS_DIAG_EMITTER_HPP
+#define TRITON_TOOLS_DIAG_EMITTER_HPP
+#include "mlir/IR/Diagnostics.h"
+#include "mlir/IR/OpDefinition.h"
+#include "triton/Tools/Sys/GetEnv.hpp"
+#include <optional>
+
+#define EMIT_PERF_WARNING(op, message)                                         \
+  if (auto out = mlir::triton::DiagnosticEmitter::getPerfWarningStream(op)) {  \
+    *out << message;                                                           \
+  }
```

Comment:
why does it need to be a macro?

Reply:
was trying to make an API supporting `emitPerfWarn(op) << "foo" << "bar"` but find it difficult to make it print nothing when the switch is off. Making an empty `InFlightDiagnostic` triggers an assertion in MLIR when it attempts to attach notes.